src/gluonts/dataset/repository/lstnet.py

```diff
@@ -0,0 +1,203 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""
+Recipes to generate electricity, traffic, exchange_rate and solar-energy dataset.
+Such dataset have been used with different processing over time by several papers such as
+`<https://arxiv.org/abs/1704.04110>`_
+`<https://arxiv.org/abs/1703.07015>`_
+`<https://arxiv.org/pdf/1509.08333>`_
+We here reuse the datasets used by LSTNet as the processed url of the datasets are available on github.
+"""
+import json
+import os
+from pathlib import Path
+from typing import List, NamedTuple, Optional
+
+import pandas as pd
+
+from gluonts.dataset.repository.util import metadata, save_to_file, to_dict
+
+
+def load_from_pandas(
+    df: pd.DataFrame,
+    time_index: pd.DatetimeIndex,
+    agg_freq: Optional[str] = None,
+) -> List[pd.Series]:
+    df = df.set_index(time_index)
+
+    pivot_df = df.transpose()
+    pivot_df.head()
+
+    timeseries = []
+    for row in pivot_df.iterrows():
+        ts = pd.Series(row[1].values, index=time_index)
+        if agg_freq is not None:
+            ts = ts.resample(agg_freq).sum()
+        first_valid = ts[ts.notnull()].index[0]
+        last_valid = ts[ts.notnull()].index[-1]
+        ts = ts[first_valid:last_valid]
+
+        timeseries.append(ts)
+
+    return timeseries
+
+
+class LstnetDataset(NamedTuple):
+    name: str
+    url: str
+    num_series: int
+    num_time_steps: int
+    prediction_length: int
+    rolling_evaluations: int
+    freq: str
+    start_date: str
+    agg_freq: Optional[str] = None
+
+
+root = "https://raw.githubusercontent.com/laiguokun/multivariate-time-series-data/master/"
+
+dataset_infos = {
```

Comment:
Maybe rename to `datasets_info` since dataset has plural but info does not.

Reply:
sure